### PR TITLE
support full table aggregations [2]

### PR DESCRIPTION
**Pull request by tantaman**
API:
```ts
q.select(agg.sum('x'), agg.avg('y'), agg.count());
```

---

Allows users to aggregate against an entire table without applying a `group-by`

The main value-add here is to change `count` to now be selectable with other fields.

E.g.,

before:

`q.count();` and `q.select(fields...)` were exclusive operations.

Now we can:

`q.select(fields..., agg.count())`

If the user selects a field that was not aggregated it returns a value from an arbitrary row.

This also allows us to drop `PrimitiveView` and make all queries sink to the same view class.

### CODE REVIEW

File: src/zql/ivm/graph/operators/full-agg-operators.test.ts [hunk 1]
```diff
@@ -0,0 +1,312 @@
...
+import {NoOp} from './operator.js';
+import {DifferenceStreamWriter} from '../difference-stream-writer.js';
+import {
+  FullAvgOperator,
+  FullCountOperator,
+  FullSumOperator,
+} from './full-agg-operators.js';
+
+test('count', () => {
+  const inputWriter = new DifferenceStreamWriter<{x: string}>();
```
Comment (tantaman) [inline src/zql/ivm/graph/operators/full-agg-operators.test.ts:12]: these commits are prior to https://github.com/rocicorp/rails/pull/55 hence the `writer` and `reader`

File: src/zql/ivm/graph/operators/full-agg-operators.ts [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+        }
+
+        if (this.#numElements + numElements === 0) {
+          this.#avg = 0;
+        } else {
+          this.#avg =
+            (this.#avg * this.#numElements + sum) /
+            (this.#numElements + numElements);
+          this.#numElements += numElements;
+        }
```
Comment (arv) [inline src/zql/ivm/graph/operators/full-agg-operators.ts:126]: I don't think you need the first branch. The second branch should compute the same result

**Review by arv — COMMENTED** [review]
LGTM

Very nice

File: src/zql/ivm/graph/operators/full-agg-operators.ts (hunk 2 above)
Comment (tantaman) [reply src/zql/ivm/graph/operators/full-agg-operators.ts:126]: second branch would divide by zero without the first :(

**Pull request merged**